include/oneapi/tbb/detail/_config.h

```diff
@@ -332,6 +332,12 @@
     #define __TBB_nodiscard
 #endif
 
+#if __clang__
+    #define __TBB_lto_visibility_public [[clang::lto_visibility_public]]
+#else
+    #define __TBB_lto_visibility_public
+#endif
+
```

Comment:
I think we need to give the macro a better, semantical name, something similar to `__TBB_EXPORTED_FUNC`. The property of a type that we want to indicate and apply as needed is that it's a "callback interface" that is declared in the library headers but used in the binaries. We have other such types, for example `delegate_base`, and so it's likely only the matter of time for similar issues with whole program optimization options to be seen for these types as well.